### PR TITLE
add vm extend disk functionality

### PR DESCRIPTION
**Pull request by mheidenr**

### CODE REVIEW

File: lib/chef/knife/vsphere_vm_extend_disk.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    vim_connection
+
+    dc = datacenter
+
+    folder = find_folder(get_config(:folder)) || dc.vmFolder
+
+    vm = find_in_folder(folder, RbVmomi::VIM::VirtualMachine, vmname) || abort("VM #{vmname} not found")
+
+    disks = vm.config.hardware.device.select do |device|
```
Comment (swalberg) [inline lib/chef/knife/vsphere_vm_extend_disk.rb]: Great use of `select`

**Comment by swalberg** [comment]
Do you know what happens if someone tries to shrink a disk, either intentionally or accidentally?

**Comment by swalberg** [comment]
What would you think about making this command `vm disk extend` instead of `vm extend disk`? I could see us having other `disk` commands like `list`. For that matter, should we drop the leading `vm`? Anyone else have thoughts on the naming?

**Comment by mheidenr** [comment]
If you try to shrink the disk you get ```RbVmomi::Fault: InvalidDeviceOperation: [...]```

**Comment by swalberg** [comment]
OK, that's cool (the InvalidDeviceOperation)... I wanted to make sure we weren't somehow going to destroy a disk if someone didn't pay attention to the units!

**Comment by mheidenr** [comment]
I agree, ```vm disk extend``` is more useful if you see this as tree of commands. As this command is only applicable to vms, you should not remove the leading ```vm```. For a more generic disk management, I would introduce a separate ```disk```command that uses the https://www.vmware.com/support/developer/vc-sdk/visdk25pubs/ReferenceGuide/vim.VirtualDiskManager.html

**Comment by mheidenr** [comment]
I changed it to ```vm disk extend```

**Comment by swalberg** [comment]
Awesome work, thanks! I'm about to follow this up with `vsphere vm disk list VMNAME`, would appreciate your comments.

**Pull request merged**